src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java

```diff
@@ -249,6 +249,31 @@ public List<MedicalWard> getMedicalsWard(char wardId, boolean stripeEmpty) throw
 		return medicalWards;
 	}
 
+	/**
+	 * Gets all the {@link Medical}s associated to Specified {@link Ward} and
+	 * Specified {@link Medical}.
+	 * 
+	 * @param wardId      the ward id.
+	 * @param medId       the medical id.
+	 * @param stripeEmpty - if {@code true}, stripes the empty lots
+	 * @return the retrieved medicals.
+	 * @throws OHServiceException if an error occurs during the medical retrieving.
+	 */
+	public List<MedicalWard> getMedicalsWard(String wardId, int medId, boolean stripeEmpty) throws OHServiceException {
+		List<MedicalWard> medicalWards = repository.findAllWhereWardAndMedical(wardId, medId);
+		for (int i = 0; i < medicalWards.size(); i++) {
+			double qty = medicalWards.get(i).getIn_quantity() - medicalWards.get(i).getOut_quantity();
+			medicalWards.get(i).setQty(qty);
+
+			if (stripeEmpty && qty == 0) {
+				medicalWards.remove(i);
+				i = i - 1;
+			}
```

Comment:
Instead of duplicating the code, why we don't have the other method calling this and just change the repository method to be called depending on `medId`?

```
public List<MedicalWard> getMedicalsWard(char wardId, boolean stripeEmpty) throws OHServiceException {
         return getMedicalsWard(String.valueOf(wardId), 0, stripeEmpty);
}

...

public List<MedicalWard> getMedicalsWard(String wardId, int medId, boolean stripeEmpty) throws OHServiceException {
           // condition on medId == 0
}
```